projects/product-configuration/public_api.ts

```diff
@@ -0,0 +1,5 @@
+/*
+ * Public API Surface of product-configuration
+ */
+
+export * from './src/product-configuration.module';
```

Comment:
Nice one. 
We should be careful about what we expose as a public api, in order to prevent breaking changes.
Let's start with only the essential stuff, such as the configuration model object, and the module itself, and slowly add other stuff as needed.